src/main/java/de/adesso/kicker/notification/NotificationService.java

```diff
@@ -1,56 +1,169 @@
     public Notification getNotificationById(long id) {
 
         return notificationRepository.findByNotificationId(id);
     }
 
-    // Get all notifications
     public List<Notification> getAllNotifications() {
 
         List<Notification> notifications = new ArrayList<>();
         notificationRepository.findAll().forEach(notifications::add);
         return notifications;
     }
 
-    // Get all notifications that were send by a specific user
     public List<Notification> getAllNotificationsBySender(User sender) {
 
         List<Notification> notifications = new ArrayList<>();
-        notificationRepository.findBySender(sender).forEach(notifications::add);
+        if (sender != null) {
+            notifications = getAllNotificationsBySender(sender.getUserId());
+        }
         return notifications;
     }
 
-    // Get all notifications that were received by a specific user
     public List<Notification> getAllNotificationsByReceiver(User receiver) {
 
+        List<Notification> notifications = new ArrayList<>();
+        if (receiver != null) {
+            notifications = getAllNotificationsByReceiver(receiver.getUserId());
+        }
+        return notifications;
+    }
+
+    public List<Notification> getAllNotificationsBySender(long senderId) {
+
+        User sender = userService.getUserById(senderId);
+        List<Notification> notifications = new ArrayList<>();
+        notificationRepository.findBySender(sender).forEach(notifications::add);
+        return notifications;
+    }
+
+    public List<Notification> getAllNotificationsByReceiver(long receiverId) {
+
+        User receiver = userService.getUserById(receiverId);
         List<Notification> notifications = new ArrayList<>();
         notificationRepository.findByReceiver(receiver).forEach(notifications::add);
         return notifications;
     }
 
-    // Save notification in repo
+    // Accepts a notification passed on type (enum)
+    public void acceptNotificationById(long id) {
+
+        Notification n = getNotificationById(id);
+        switch (n.getType()) {
+        case Notification:
+            removeNotificationById(id);
+            break;
+        case TeamJoinRequest:
+            teamJoinRequestService.acceptTeamJoinRequest(id);
+            removeNotificationById(id);
+            break;
+        case MatchCreationRequest:
+            matchCreationRequestService.acceptMatchJoinRequest(id);
+            break;
+        case TournamentJoinRequest:
+            tournamentJoinRequestService.acceptTournamentJoinRequest(id);
+            removeNotificationById(id);
+            break;
+        case MatchVerificationRequest:
+            matchVerificationRequestService.acceptMatchVerificationRequest(id);
+            removeNotificationById(id);
+            break;
+        default:
+            removeNotificationById(id);
+        }
+    }
+
     public void saveNotification(Notification notification) {
+
+        if (notification == null) {
```

Comment:
All these null checks are unnecessary as users can't manually enter values when notifications are used. You did this all over the code and makes is unnecessarily more difficult to read.